torch/csrc/jit/codegen/cuda/lower_mma.cpp

```diff
@@ -0,0 +1,609 @@
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_ir_builder.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+#include <torch/csrc/jit/codegen/cuda/lower_mma.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
+#include <torch/csrc/jit/codegen/cuda/scheduler/mma_utils.h>
+
+#include <sstream>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+namespace {
+
+class LowerMMA {
+ public:
+  static std::vector<kir::Expr*> run(std::vector<kir::Expr*> exprs) {
+    LowerMMA lower_mma;
+    return lower_mma.handle(exprs);
+  }
+
+ private:
+  std::vector<kir::Expr*> handle(const std::vector<kir::Expr*>& exprs) {
```

Comment:
Wondering if a fused multiply-sum op would make this significantly easier or not.